xarray/core/dataset.py

```diff
@@ -962,6 +962,11 @@ def _replace_vars_and_dims(
         Unlike _replace_with_new_dims(), this method always recalculates
         indexes from variables.
         """
+        warnings.warn(
+            "`_replace_vars_and_dims` is deprecated in favor of `_replace_with_new_dims` and will be removed in 0.16",
+            DeprecationWarning,
+            stacklevel=2,
+        )
```

Comment:
Do we really need to deprecate this? It's an internal API, so I think we should feel free to remove it without warning.

Reply:
Yeah, I wasn't sure, and thought it safer to deprecate, but I was probably being overly cautious. I'll remove.